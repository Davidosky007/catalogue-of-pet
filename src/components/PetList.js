import PropTypes from 'prop-types';
import Pet from './Pet';

const PetsList = ({ pets }) => (
  <div className="pets__wrapper" data-testid="pets-list-id">
    {pets.length === 0 ? (
      <h1 className="loading__wrapper">No pets Found</h1>
    ) : (
      pets.map((pet) => (
        <Pet id={pet.id} name={pet.name} key={pet.id} media={pet.photos} />
      ))
    )}
  </div>
);

PetsList.propTypes = {
  pets: PropTypes.arrayOf(PropTypes.any),
};

PetsList.defaultProps = {
  pets: [],
};

export default PetsList;
