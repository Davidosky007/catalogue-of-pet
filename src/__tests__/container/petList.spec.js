import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import PetList from '../../components/PetList';

test('should render Details components properly ', () => {
  const details = renderer.create(<PetList />);
  const final = details.toJSON();
  expect(final).toMatchSnapshot();
});
