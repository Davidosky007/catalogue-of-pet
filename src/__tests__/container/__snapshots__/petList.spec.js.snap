// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details components properly  1`] = `
<div
  className="pets__wrapper"
  data-testid="pets-list-id"
>
  <h1
    className="loading__wrapper"
  >
    No pets Found
  </h1>
</div>
`;
